frontend/src/services/web3-provider/types.ts

```diff
@@ -32,10 +29,26 @@ interface RequestArguments {
   readonly params?: readonly unknown[] | object;
 }
 
-export type Eip1193Provider = ExternalProvider & {
+export interface DisconnectableProvider {
+  disconnect(): Promise<void>;
+}
+
+export type ExternalProvider = Eip1193Provider & {
+  isMetaMask?: boolean;
+  sendAsync?: (
+    request: RequestArguments,
+    callback: (error: unknown, response: unknown) => void,
+  ) => void;
+  send?: (
+    request: RequestArguments,
+    callback: (error: unknown, response: unknown) => void,
+  ) => void;
+  request?: (request: RequestArguments) => Promise<unknown>;
+};
```

Comment:
Nitpick: You are you using a different syntax here for the function declarations than in the other types. Would be nice to be consistent at least in this file.
e.g. you should instead write `send?(request: RequestArguments, callback: (error: unknown, response: unknown) => void): void`.

Besides, the `request` function should not get defined a second time here.

Reply:
Good point!